checkstyle/src/main.rs

```diff
@@ -0,0 +1,133 @@
+use regex::Regex;
+use std::fs::{self};
+// use std::{fs::{self}, io::BufRead};
+use text_io::read;
+// use std::path::Path;
+// use std::path::PathBuf;
+// use std::fs::File;
+// use std::io::prelude::*;
+
+fn get_file(filename: &str) -> String {
+    fs::read_to_string(filename).expect("Failed to read file")
+}
+
+fn split_into_lines(regex: &str) -> Vec<&str> {
+    regex.lines().collect()
+}
+
+fn check_matching_regex_line(file: &str, regex_line: &str) -> Option<bool> {
+    let re = Regex::new(regex_line).ok()?; // Use .ok() to convert Result to Option
```

Comment:
Let's pass in an already constructed regex to this function and have it just return `bool`. We can construct the regex outside and handle the error of an invalid regex there.